sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java

```diff
@@ -337,4 +346,26 @@ public EmptyPagesFilterTransformer(
                 initialContinuationToken,
                 top);
     }
+
+    private int fluxSequentialMergeConcurrency(FeedOptions options, int numberOfPartitions) {
+        int parallelism = options.getMaxDegreeOfParallelism();
+        if (parallelism < 0) {
+            parallelism = Configs.CPU_CNT;
+        } else if (parallelism == 0) {
+            parallelism = 1;
+        }
+
+        return Math.min(numberOfPartitions, parallelism);
+    }
+
+    private int fluxSequentialMergePrefetch(FeedOptions options, int numberOfPartitions, int pageSize, int fluxConcurrency) {
+        int maxBufferedItemCount = options.getMaxBufferedItemCount();
+
+        if (maxBufferedItemCount <= 0) {
+            maxBufferedItemCount = Math.min(Configs.CPU_CNT * numberOfPartitions * pageSize, 100_000);
```

Comment:
Any particular reason to use 100_000 ?

Reply:
This is kind of heuristic but general idea is that if there are 100_000 documents of size 1KB, that means 100_000*1KB=100MB in memory which will start to put pressure on memory if we go beyond that.